src/types.ts

```diff
@@ -104,11 +104,12 @@ export interface Features {
 }
 
 /**
- * Function which transforms the EvaluationContext to a type useful for the provider.
+ * Interface that providers must implement to resolve flag values for their particular
+ * backend or vendor.
+ *
+ * Implementation for resolving all the required flag types must be defined.
  */
-export type ContextTransformer<T = unknown> = (context: EvaluationContext) => T;
-
-interface GenericProvider<T> {
+export interface Provider extends Pick<Partial<EvaluationLifeCycle>, 'hooks'> {
```

Comment:
@beeme1mr this is the way I've represented the hooks on the provider. I chose this for a few reasons...

We don't want any one to be able to ADD hooks, so I've just added a hook property getter... then the hook author can add hooks at the point they define their provider. We don't want anyone else attaching hooks with an `addHooks()` method, or clearning them with `clearHooks()`.

I also don't want to use a base class, interfaces are always more flexible.

Reply:
I approve